lib/rspec/support/method_signature_verifier.rb

```diff
@@ -187,13 +187,19 @@ def classify_parameters
       if Java::JavaLang::String.instance_method(:char_at).arity == -1
         class MethodSignature < remove_const(:MethodSignature)
         private
+          def java_instance_methods
+            if @method.owner.java_class.public?
+              [@method.owner.java_class]
+            else
+              @method.owner.java_class.interfaces + [@method.owner.java_class.generic_superclass]
+            end.select(&:public?).map(&:java_instance_methods).flatten
+          end
```

Comment:
Could we refactor this back inline below?
```
java_instance_methods =
  if @method.owner.java_class.public?
    @method.owner.java_class.java_instance_methods
  else
    (
      @method.owner.java_class.interfaces +
      [@method.owner.java_class.generic_superclass]
    ).inject([]) do |methods, klass|
      return methods unless klass.public? 
      methods += klass.java_instance_methods
  end
```

Reply:
rubocop will complain about ABC size if I put this inline.  What do you suggest?

Reply:
Hm ok, maybe refactor this as a seperate method then, I'd like to see an early return without any looping if the class is public